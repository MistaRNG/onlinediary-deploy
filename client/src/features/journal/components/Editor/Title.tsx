import { TextArea } from "@blueprintjs/core";
import React from "react";

interface TitleProps {
  titleRef: React.RefObject<HTMLTextAreaElement>;
  keyUpHandler: (event: React.KeyboardEvent<HTMLTextAreaElement>) => void;
  keyDownHandler: (event: React.KeyboardEvent<HTMLTextAreaElement>) => void;
}

const Title: React.FC<TitleProps> = ({ titleRef, keyUpHandler, keyDownHandler }) => {
  return (
    <TextArea
      large={true}
      placeholder="Title"
      maxLength={60}
      inputRef={titleRef}
      onKeyDown={keyDownHandler}
      onKeyUp={keyUpHandler}
    />
  );
};

export default Title;
