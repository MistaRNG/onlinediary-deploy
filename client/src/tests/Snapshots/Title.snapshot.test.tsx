import React from 'react';
import { render } from '@testing-library/react';
import Title from '../../features/journal/components/Editor/Title';
import '@testing-library/jest-dom';
import { describe, it, expect, jest } from '@jest/globals';

describe('Title Component Snapshot Tests', () => {
  it('should render the Title component correctly', () => {
    // ARRANGE
    const mockTitleRef = React.createRef<HTMLTextAreaElement>();
    const mockKeyUpHandler = jest.fn();
    const mockKeyDownHandler = jest.fn();

    // ACT
    const { asFragment } = render(
      <Title
        titleRef={mockTitleRef}
        keyUpHandler={mockKeyUpHandler}
        keyDownHandler={mockKeyDownHandler}
      />
    );

    // ASSERT
    expect(asFragment()).toMatchSnapshot();
  });
});
