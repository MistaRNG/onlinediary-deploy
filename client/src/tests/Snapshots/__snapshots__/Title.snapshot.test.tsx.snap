// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component Snapshot Tests should render the Title component correctly 1`] = `
<DocumentFragment>
  <textarea
    class="bp4-input bp4-large"
    maxlength="60"
    placeholder="Title"
  />
</DocumentFragment>
`;
